lib/distribute/writers/csv-bal-with-lang.cjs

```diff
@@ -0,0 +1,49 @@
+/* eslint camelcase: off */
+const {promisify} = require('util')
+const {join} = require('path')
+const {createWriteStream} = require('fs')
+const {createGzip} = require('zlib')
+const finished = promisify(require('stream').finished)
+const {ensureDir} = require('fs-extra')
+const csvWriter = require('csv-write-stream')
+const pumpify = require('pumpify')
+
+const {prepareAdresses} = require('../../formatters/csv-bal.cjs')
+
+function createCsvWriteStream(file) {
+  return pumpify.obj(
+    csvWriter({separator: ';'}),
+    createGzip(),
+    file
+  )
+}
+
+function waitForDrain(stream) {
+  if (stream.writableLength > stream.writableHighWaterMark) {
+    return new Promise(resolve => {
+      stream.once('drain', resolve)
```

Comment:
Je ne suis pas certain de comprendre l'usage de 'drain' ici. Si tu as 2min, je suis preneur pour un éclaircissement. 😅

Reply:
Pour le coup, c'est un copié collé de `lib/distribute/writers/csv-bal.cjs`, l'explication est assez complexe https://stackoverflow.com/questions/18932488/how-to-use-drain-event-of-stream-writable-in-node-js, j'ai l'impression que ca permet d'attendre la fin de l'écriture même si on dépasse la taille maximum du buffer d'écriture.